index/miner/meta.go

```diff
@@ -45,8 +45,10 @@ func (mi *Index) metaWorker() {
 			// doing a merge. Will depend if this too slow, but might not be the case
 			mi.lock.Lock()
 			addrs := make([]string, 0, len(mi.index.Chain.Power))
-			for addr := range mi.index.Chain.Power {
-				addrs = append(addrs, addr)
+			for addr, power := range mi.index.Chain.Power {
+				if power.Power > 0 {
+					addrs = append(addrs, addr)
+				}
```

Comment:
Same idea as in the ask index.
We only try to solve miner metadata (geography, is online or not, etc) if the miner has some power.
Saves tons of work and delays with miners that doesn't really exists. (And by definition since have no power, can't make deals with).